base/test/h264encoderv4l2_tests.cpp

```diff
@@ -212,4 +218,114 @@ BOOST_AUTO_TEST_CASE(encodepush, *boost::unit_test::disabled())
 	LOG_INFO << "TEST DONE";
 }
 
+BOOST_AUTO_TEST_CASE(encode_and_extract_motion_vectors)
+{
+	// metadata is known
+	auto width = 640;
+	auto height = 360;
+
+	auto fileReader = boost::shared_ptr<FileReaderModule>(new FileReaderModule(FileReaderModuleProps("./data/Raw_YUV420_640x360/Image???_YUV420.raw")));
+	auto metadata = framemetadata_sp(new RawImagePlanarMetadata(width, height, ImageMetadata::ImageType::YUV420, size_t(0), CV_8U));
+	auto rawImagePin = fileReader->addOutputPin(metadata);
+
+	H264EncoderV4L2Props encoderProps;
+	encoderProps.targetKbps = 1024;
+	encoderProps.enableMotionVectors = true;
+	auto encoder = boost::shared_ptr<Module>(new H264EncoderV4L2(encoderProps));
+	fileReader->setNext(encoder);
+
+	auto sink = boost::shared_ptr<ExternalSinkModule>(new ExternalSinkModule());
+	encoder->setNext(sink);
+
+	BOOST_TEST(fileReader->init());
+	BOOST_TEST(encoder->init());
+	BOOST_TEST(sink->init());
+
+	fileReader->play(true);
+
+	for (auto i = 0; i < 40; i++)
+	{
+		fileReader->step();
+		encoder->step();
+		auto frames = sink->pop();
+		for (auto it = frames.cbegin(); it != frames.cend(); it++)
```

Comment:
add a check for count of frames for which you got motion data - outside of the for loop

Reply:
done